tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/FilterStaticFrameworksTaskTest.cs

```diff
@@ -0,0 +1,225 @@
+				0x03, 0x00, 0x00, 0x00,
+				// File type (MH_DYLIB = 6)
+				0x06, 0x00, 0x00, 0x00,
+				// Number of load commands
+				0x00, 0x00, 0x00, 0x00,
+				// Size of load commands
+				0x00, 0x00, 0x00, 0x00,
+				// Flags
+				0x00, 0x00, 0x00, 0x00,
+				// Reserved (64-bit only)
+				0x00, 0x00, 0x00, 0x00
+			};
+			return header;
+		}
+
+		[Test]
+		[TestCase (ApplePlatform.iOS, "libavcodec.framework", "libavcodec.dylib")]
+		[TestCase (ApplePlatform.TVOS, "libavcodec.framework", "libavcodec.dylib")]
+		[TestCase (ApplePlatform.MacOSX, "libavcodec.framework", "libavcodec.dylib")]
+		[TestCase (ApplePlatform.MacCatalyst, "libavcodec.framework", "libavcodec.dylib")]
+		public void TestCustomFrameworkExecutablePath (ApplePlatform platform, string frameworkName, string executableName)
+		{
+			TestCustomFrameworkExecutablePathForPlatform (platform, frameworkName, executableName);
+		}
+
+		void TestCustomFrameworkExecutablePathForPlatform (ApplePlatform platform, string frameworkName, string executableName)
+		{
+			var isDesktopPlatform = platform == ApplePlatform.MacOSX || platform == ApplePlatform.MacCatalyst;
+
+			// Arrange: Create a mock framework with custom CFBundleExecutable
+			var tempDir = Cache.CreateTemporaryDirectory ();
+			var frameworkDir = Path.Combine (tempDir, platform.AsString (), frameworkName);
+			Directory.CreateDirectory (frameworkDir);
+
+			string infoPlistPath;
+			if (isDesktopPlatform) {
+				// macOS and MacCatalyst structure: Framework.framework/Versions/A/Resources/Info.plist
+				var versionsDir = Path.Combine (frameworkDir, "Versions", "A");
+				var resourcesDir = Path.Combine (versionsDir, "Resources");
+				Directory.CreateDirectory (resourcesDir);
+				infoPlistPath = Path.Combine (resourcesDir, "Info.plist");
+
+				// Create symlinks as they exist in real frameworks
+				Directory.CreateSymbolicLink (Path.Combine (frameworkDir, "Resources"), "Versions/A/Resources");
+			} else {
+				// iOS and tvOS structure: Framework.framework/Info.plist
+				infoPlistPath = Path.Combine (frameworkDir, "Info.plist");
+			}
+
+			// Create Info.plist with custom CFBundleExecutable
+			var infoPlistContent = @"<?xml version=""1.0"" encoding=""UTF-8""?>
+<!DOCTYPE plist PUBLIC ""-//Apple//DTD PLIST 1.0//EN"" ""http://www.apple.com/DTDs/PropertyList-1.0.dtd"">
+<plist version=""1.0"">
+<dict>
+	<key>CFBundleExecutable</key>
+	<string>" + executableName + @"</string>
+	<key>CFBundleIdentifier</key>
+	<string>com.ffmpeg.libavcodec</string>
+</dict>
+</plist>";
+			File.WriteAllText (infoPlistPath, infoPlistContent);
+
+			// Create the custom executable file 
+			var customExecutablePath = Path.Combine (frameworkDir, executableName);
+			File.WriteAllBytes (customExecutablePath, CreateMinimalMachODylib ());
+
+			// Act: Create and execute the task
+			var task = CreateTask<FilterStaticFrameworks> ();
+			task.FrameworkToPublish = new ITaskItem [] { new TaskItem (frameworkDir) };
+			task.OnlyFilterFrameworks = true;
```

Comment:
Set the TargetFrameworkMoniker property on the task to "TargetFramework.GetTargetFramework (platform).ToString ();" on every task creation in this file.